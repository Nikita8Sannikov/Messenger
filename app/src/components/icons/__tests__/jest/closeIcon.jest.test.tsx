import { render, screen } from "@testing-library/react";

import CloseIconComponent from "@components/icons/closeIcon";

describe("ArrowLeftIcon", () => {
    test("renders correctly", () => {
        render(<CloseIconComponent />);
        const icon = screen.getByTestId("close-icon");
        expect(icon).toBeInTheDocument();
    });

    test("renders with custom size", () => {
        render(<CloseIconComponent size={32} />);
        const icon = screen.getByTestId("close-icon");
        expect(icon).toHaveAttribute("width", "32px");
        expect(icon).toHaveAttribute("height", "32px");
    });

    test("uses string size as is", () => {
        render(<CloseIconComponent size="2rem" />);
        const icon = screen.getByTestId("close-icon");
        expect(icon).toHaveAttribute("width", "2rem");
        expect(icon).toHaveAttribute("height", "2rem");
    });

    test("matches snapshot", () => {
        const { asFragment } = render(<CloseIconComponent></CloseIconComponent>);

        // Снимаем снапшот компонента
        expect(asFragment()).toMatchSnapshot();
    });
});