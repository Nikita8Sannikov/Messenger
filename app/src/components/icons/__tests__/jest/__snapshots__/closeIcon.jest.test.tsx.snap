// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowLeftIcon matches snapshot 1`] = `
<DocumentFragment>
  <svg
    data-testid="close-icon"
    height="24px"
    viewBox="0 0 24 24"
    width="24px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M18 6L6 18M6 6l12 12"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</DocumentFragment>
`;
